pkg/operation/botanist/addons/nginxingress/nginxingress.go

```diff
@@ -0,0 +1,699 @@
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+					labelKeyComponent:         labelValueController,
+				},
+				Namespace: n.namespace,
+			},
+			Data: n.values.ConfigData,
+		}
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceAccountName,
+				Namespace: n.namespace,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp:         labelAppValue,
+					labelKeyRelease:                   labelValueAddons,
+					"addonmanager.kubernetes.io/mode": "Reconcile",
+				},
+			},
+			AutomountServiceAccountToken: pointer.Bool(false),
+		}
+
+		clusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterRoleName,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+				},
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"endpoints", "nodes", "pods", "secrets"},
+					Verbs:     []string{"list", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"nodes"},
+					Verbs:     []string{"get"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"services", "configmaps"},
+					Verbs:     []string{"get", "list", "update", "watch"},
+				},
+				{
+					APIGroups: []string{"extensions", "networking.k8s.io"},
+					Resources: []string{"ingresses"},
+					Verbs:     []string{"get", "list", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"events"},
+					Verbs:     []string{"create", "patch"},
+				},
+				{
+					APIGroups: []string{"extensions", "networking.k8s.io"},
+					Resources: []string{"ingresses/status"},
+					Verbs:     []string{"update"},
+				},
+				{
+					APIGroups: []string{"networking.k8s.io"},
+					Resources: []string{"ingressclasses"},
+					Verbs:     []string{"get", "list", "watch"},
+				},
+				{
+					APIGroups: []string{"coordination.k8s.io"},
+					Resources: []string{"leases"},
+					Verbs:     []string{"list", "watch"},
+				},
+			},
+		}
+
+		clusterRoleBinding = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterRoleBindingName,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+				},
+				Annotations: map[string]string{resourcesv1alpha1.DeleteOnInvalidUpdate: "true"},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		deploymentBackend = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deploymentNameBackend,
+				Namespace: n.namespace,
```

Comment:
```suggestion
				Namespace: addonNamespace,
```
nginx-ingress-addon should be deployed in `kube-system` namespace of the shoot.